src/components/documentManager.ts

```diff
@@ -0,0 +1,678 @@
+     * @param {BaseItemDto} item to get backdrop from
+     */
+    public static setPlayerBackdrop(item: BaseItemDto): void {
+        // no backdrop rotation for cc audio
+        if (getActiveDeviceId() === deviceIds.AUDIO) return;
+
+        let backdropUrl: string | null = null;
+
+        if (item.BackdropImageTags && item.BackdropImageTags.length) {
+            backdropUrl = JellyfinApi.createUrl(
+                `Items/${item.Id}/Images/Backdrop/0?tag=${item.BackdropImageTags[0]}`
+            );
+        } else if (
+            item.ParentBackdropItemId &&
+            item.ParentBackdropImageTags &&
+            item.ParentBackdropImageTags.length
+        ) {
+            backdropUrl = JellyfinApi.createUrl(
+                `Items/${item.ParentBackdropItemId}/Images/Backdrop/0?tag=${item.ParentBackdropImageTags[0]}`
+            );
+        }
+
+        if (backdropUrl != null) {
+            window.mediaElement?.style.setProperty(
+                '--background-image',
+                `url("${backdropUrl}")`
+            );
+        } else {
+            window.mediaElement?.style.removeProperty('--background-image');
+        }
+    }
+    /* /BACKDROP LOGIC */
+
+    /**
+     * Set the URL to the item logo, or null to remove it
+     *
+     * @param {string | null} src Source url or null
+     */
+    public static setLogo(src: string | null): void {
+        const element: HTMLElement | null = document.querySelector(
+            '.detailLogo'
+        );
+        if (element) {
+            this.setBackgroundImage(element, src);
+        } else {
+            console.error('documentManager: Cannot find .detailLogo');
+        }
+    }
+
+    /**
+     * Set the URL to the item banner image (I think?),
+     * or null to remove it
+     *
+     * @param {string | null} src Source url or null
+     */
+    public static setDetailImage(src: string | null): void {
+        const element: HTMLElement | null = document.querySelector(
+            '.detailImage'
+        );
+        if (element) {
+            this.setBackgroundImage(element, src);
+        } else {
+            console.error('documentManager: Cannot find .detailImage');
+        }
+    }
+
+    /**
+     * Set the human readable name for an item
+     *
+     * This combines the old statement setDisplayName(getDisplayName(item))
+     * into setDisplayName(item).
+     *
+     * @param {BaseItemDto} item source for the displayed name
+     */
+    private static setDisplayName(item: BaseItemDto): void {
+        const name: string = item.EpisodeTitle ?? <string>item.Name;
+
+        let displayName: string = name;
+
+        if (item.Type == 'TvChannel') {
+            if (item.Number) displayName = `${item.Number} ${name}`;
+        } else if (
+            item.Type == 'Episode' &&
+            item.IndexNumber != null &&
+            item.ParentIndexNumber != null
+        ) {
+            let episode = `S${item.ParentIndexNumber}, E${item.IndexNumber}`;
```

Comment:
Unrelated, but we should remember about this so we translate it once #111 is merged